gapis/api/vulkan/minimize_textures.go

```diff
@@ -0,0 +1,113 @@
+	ctx = log.Enter(ctx, "Minimize textures")
+	return transform.Transform("Minimize textures", func(ctx context.Context,
+		id api.CmdID, cmd api.Cmd, out transform.Writer) {
+
+		const newTexWidth = 1
+		const newTexHeight = 1
+		const newTexDepth = 1
+		const regionCount = 1
+
+		s := out.State()
+		cb := CommandBuilder{Thread: cmd.Thread(), Arena: s.Arena}
+		cmd.Extras().Observations().ApplyReads(s.Memory.ApplicationPool())
+		switch cmd := cmd.(type) {
+		case *VkCreateImage:
+			imageCreateInfo := cmd.PCreateInfo().MustRead(ctx, cmd, s, nil)
+			if 0 != (imageCreateInfo.Usage() & VkImageUsageFlags(VkImageUsageFlagBits_VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT)) {
+				out.MutateAndWrite(ctx, id, cmd)
+				break
+			}
+
+			imageCreateInfo.SetExtent(NewVkExtent3D(s.Arena, newTexWidth, newTexHeight, newTexDepth))
+
+			imageCreateInfoData := s.AllocDataOrPanic(ctx, imageCreateInfo)
+			defer imageCreateInfoData.Free()
+
+			newCmd := cb.VkCreateImage(
+				cmd.Device(),
+				imageCreateInfoData.Ptr(),
+				cmd.PAllocator(),
+				cmd.PImage(),
+				VkResult_VK_SUCCESS,
+			).AddRead(imageCreateInfoData.Data())
+
+			for _, w := range cmd.Extras().Observations().Writes {
+				newCmd.AddWrite(w.Range, w.ID)
+			}
+			out.MutateAndWrite(ctx, id, newCmd)
+		case *VkCmdCopyBufferToImage:
+			bufferImageCopy := cmd.PRegions().MustRead(ctx, cmd, s, nil)
+			bufferImageCopy.SetImageOffset(NewVkOffset3D(s.Arena, 0, 0, 0))
+			imageExtent := bufferImageCopy.ImageExtent()
+			width := u32.Min(imageExtent.Width(), newTexWidth)
+			height := u32.Min(imageExtent.Height(), newTexHeight)
+			depth := u32.Min(imageExtent.Depth(), newTexDepth)
+
+			bufferImageCopy.SetImageExtent(NewVkExtent3D(s.Arena, width, height, depth))
+			bufferImageCopyData := s.AllocDataOrPanic(ctx, bufferImageCopy)
+			defer bufferImageCopyData.Free()
+
+			newCmd := cb.VkCmdCopyBufferToImage(
+				cmd.commandBuffer,
+				cmd.srcBuffer,
+				cmd.dstImage,
+				cmd.dstImageLayout,
+				regionCount,
+				bufferImageCopyData.Ptr(),
+			).AddRead(bufferImageCopyData.Data())
+
+			out.MutateAndWrite(ctx, id, newCmd)
+		case *VkCmdCopyImageToBuffer:
+			bufferImageCopy := cmd.PRegions().MustRead(ctx, cmd, s, nil)
+			bufferImageCopy.SetImageOffset(NewVkOffset3D(s.Arena, 0, 0, 0))
+			imageExtent := bufferImageCopy.ImageExtent()
+			width := u32.Min(imageExtent.Width(), newTexWidth)
+			height := u32.Min(imageExtent.Height(), newTexHeight)
+			depth := u32.Min(imageExtent.Depth(), newTexDepth)
+
+			bufferImageCopy.SetImageExtent(NewVkExtent3D(s.Arena, width, height, depth))
+			bufferImageCopyData := s.AllocDataOrPanic(ctx, bufferImageCopy)
+			defer bufferImageCopyData.Free()
+
+			newCmd := cb.VkCmdCopyImageToBuffer(
+				cmd.commandBuffer,
+				cmd.srcImage,
+				cmd.srcImageLayout,
+				cmd.dstBuffer,
+				regionCount,
+				bufferImageCopyData.Ptr(),
+			).AddRead(bufferImageCopyData.Data())
+
+			out.MutateAndWrite(ctx, id, newCmd)
+		default:
```

Comment:
We should also handle `VkCmdBlitImage` `VkCmdCopyImage` 
If the image can/would be used as a framebuffer, this may cause problems as well. Likely we should check for `VkImageUsageFlagBits_VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT` and do nothing, since that would indicate this is a color attachment.